libs/components/src/lib/tag/tag.spec.ts

```diff
@@ -0,0 +1,311 @@
+import { elementUpdated, fixture, getBaseElement } from '@vivid-nx/shared';
+import { FoundationElementRegistry } from '@microsoft/fast-foundation';
+import { Icon } from '../icon/icon';
+import { Tag } from './tag';
+import { tagDefinition } from './definition';
+import '.';
+
+const COMPONENT_TAG = 'vwc-tag';
+const ICON_SELECTOR = 'vwc-icon';
+
+/**
+ * @param element
+ * @param removable
+ */
+async function toggleRemovable(element: Tag, removable = true) {
+	element.removable = removable;
+	await elementUpdated(element);
+}
+
+/**
+ * @param element
+ * @param removable
+ */
+async function toggleSelectable(element: Tag, selectable = true) {
+	element.selectable = selectable;
+	await elementUpdated(element);
+}
+
+
+describe('vwc-tag', () => {
+	let element: Tag;
+
+	beforeEach(async () => {
+		element = (await fixture(`<${COMPONENT_TAG}></${COMPONENT_TAG}>`)) as Tag;
+	});
+
+	describe('basic', () => {
+		it('initializes as a vwc-tag', async () => {
+			expect(tagDefinition()).toBeInstanceOf(FoundationElementRegistry);
+			expect(element).toBeInstanceOf(Tag);
+			expect(element.label).toEqual(undefined);
+			expect(element.icon).toBeUndefined();
+			expect(element.connotation).toBeUndefined();
+			expect(element.shape).toBeUndefined();
+			expect(element.appearance).toBeUndefined();
+		});
+	});
+
+	describe('icon', () => {
+		it('adds an icon to the tag', async () => {
+			element.icon = 'home';
+			await elementUpdated(element);
+
+			const icon = element.shadowRoot?.querySelector(ICON_SELECTOR) as Icon;
+			expect(icon)
+				.toBeInstanceOf(Icon);
+			expect(icon?.name)
+				.toEqual('home');
+		});
+	});
+
+	describe('label', () => {
+		it('set label property to node', async () => {
+			const label = 'lorem';
+			element.label = label;
+			await elementUpdated(element);
+
+			const base = element.shadowRoot?.querySelector('.base');
+			expect(base?.textContent?.trim())
+				.toEqual(label);
+		});
+	});
+
+	describe('connotation', () => {
+		it('sets correct internal connotation style', async () => {
+			const connotation = 'cta';
+			(element as any).connotation = connotation;
+			await elementUpdated(element);
+
+			const base = element.shadowRoot?.querySelector(`.base.connotation-${connotation}`);
+			expect(base)
+				.toBeInstanceOf(Element);
```

Comment:
I'd expect this to be more explicit:
```suggestion
			const baseElementContainsConnotationClass = getBaseElement(element).classList.contains(`connotation-${connotation}`;
			expect(baseElementContainsConnotationClass)
				.toBeTruthy();
```